tests/runtime/strings/wasm_utf16.c

```diff
@@ -0,0 +1,31 @@
+#include <assert.h>
+#include <imports.h>
+#include <exports.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+
+char16_t BASIC_STRING[] = u"latin utf16";
+char16_t STR_BUFFER[500];
+
+void exports_test_imports() {
+  imports_string_t str1;
+  imports_string_set(&str1, BASIC_STRING);
+  imports_f1(&str1);
+  imports_string_t str2;
+  imports_f2(&str2);
+  memcpy(&STR_BUFFER[0], str2.ptr, str2.len * 2);
+  STR_BUFFER[str2.len] = '\0';
+}
+
+void exports_f1(exports_string_t *str) {
+  assert(str->len > 0);
+  memcpy(&STR_BUFFER[0], str->ptr, str->len * 2);
+  STR_BUFFER[str->len] = '\0';
+}
+
+void exports_f2(exports_string_t *ret) {
+  exports_string_set(ret, STR_BUFFER);
+  assert(ret->len > 0);
+  assert(memcmp(ret->ptr, STR_BUFFER, ret->len * 2) == 0);
```

Comment:
Sorry I meant moreso there are no assertions that a particular value is actually received in C. The assertions added here aren't too meaningful of the actual bytes received, e.g. that the endianness was all correct. Could a function be added to the C exports where it receives a known string and then asserts that it receives that string?

Reply:
Ah I see what you mean, sure.

Reply:
I've added some direct assertions in the import pattern. The other pattern is a write / read pattern from the JS side so there isn't a reliable assertion to run unless we track the call count.